### PR TITLE
Benchmarks for stream limits, combine tests and benchmarks into fewer files

### PR DESCRIPTION
**Pull request by mprimi**
- [ ] Link to issue, e.g. `Resolves #NNN`
 - [ ] Documentation added (if applicable)
 - [x] Tests added
 - [x] Branch rebased on top of current main (`git pull --rebase origin main`)
 - [x] Changes squashed to a single commit (described [here](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html))
 - [x] Build is green in Travis CI
 - [x] You have certified that the contribution is your original work and that you license the work to the project under the [Apache 2 license](https://github.com/nats-io/nats-server/blob/main/LICENSE)

### Changes proposed in this pull request:

 - Add JetStream benchmark that measures publish throughput with different limits (MaxBytes, MaxMessages,  MaxPerSubject, MaxAge, ...)
 - Merge `jetstream_chaos_*_test.go` and and `jetstream_benchmark_*_test.go` into two files (down from 8)

Example output (filtered subset):

```
$ go test -v -bench 'BenchmarkJetStreamInterestStreamWithLimit/.*R=1.*/Storage=Memory/.*' -run NONE -benchtime 200000x -count 1

goos: darwin
goarch: arm64
pkg: github.com/nats-io/nats-server/v2/server
BenchmarkJetStreamInterestStreamWithLimit
    jetstream_benchmark_interest_stream_limit_test.go:44: BatchSize: 100, MsgSize: 256, Subjects: 2500, Publishers: 4, Random Message: true
BenchmarkJetStreamInterestStreamWithLimit/N=1,R=1
BenchmarkJetStreamInterestStreamWithLimit/N=1,R=1/Storage=Memory
BenchmarkJetStreamInterestStreamWithLimit/N=1,R=1/Storage=Memory/unlimited
    jetstream_benchmark_interest_stream_limit_test.go:230: Stream: {clusterSize:1 replicas:1}, Storage: [Memory] Limit: [unlimited], Ops: 200000
BenchmarkJetStreamInterestStreamWithLimit/N=1,R=1/Storage=Memory/unlimited-8              200000              9743 ns/op          26.28 MB/s             0 %error
BenchmarkJetStreamInterestStreamWithLimit/N=1,R=1/Storage=Memory/MaxMsg=1000
    jetstream_benchmark_interest_stream_limit_test.go:230: Stream: {clusterSize:1 replicas:1}, Storage: [Memory] Limit: [MaxMsg=1000], Ops: 200000
BenchmarkJetStreamInterestStreamWithLimit/N=1,R=1/Storage=Memory/MaxMsg=1000-8            200000              9800 ns/op          26.12 MB/s             0 %error
BenchmarkJetStreamInterestStreamWithLimit/N=1,R=1/Storage=Memory/MaxMsg=10
    jetstream_benchmark_interest_stream_limit_test.go:230: Stream: {clusterSize:1 replicas:1}, Storage: [Memory] Limit: [MaxMsg=10], Ops: 200000
BenchmarkJetStreamInterestStreamWithLimit/N=1,R=1/Storage=Memory/MaxMsg=10-8              200000              9717 ns/op          26.35 MB/s             0 %error
BenchmarkJetStreamInterestStreamWithLimit/N=1,R=1/Storage=Memory/MaxPerSubject=10
    jetstream_benchmark_interest_stream_limit_test.go:230: Stream: {clusterSize:1 replicas:1}, Storage: [Memory] Limit: [MaxPerSubject=10], Ops: 200000
BenchmarkJetStreamInterestStreamWithLimit/N=1,R=1/Storage=Memory/MaxPerSubject=10-8       200000             78796 ns/op           3.25 MB/s             0 %error
BenchmarkJetStreamInterestStreamWithLimit/N=1,R=1/Storage=Memory/MaxAge=1s
    jetstream_benchmark_interest_stream_limit_test.go:230: Stream: {clusterSize:1 replicas:1}, Storage: [Memory] Limit: [MaxAge=1s], Ops: 200000
BenchmarkJetStreamInterestStreamWithLimit/N=1,R=1/Storage=Memory/MaxAge=1s-8              200000              9648 ns/op          26.53 MB/s             0 %error
BenchmarkJetStreamInterestStreamWithLimit/N=1,R=1/Storage=Memory/MaxBytes=1MB
    jetstream_benchmark_interest_stream_limit_test.go:230: Stream: {clusterSize:1 replicas:1}, Storage: [Memory] Limit: [MaxBytes=1MB], Ops: 200000
BenchmarkJetStreamInterestStreamWithLimit/N=1,R=1/Storage=Memory/MaxBytes=1MB-8           200000              9706 ns/op          26.38 MB/s             0 %error
PASS
ok      github.com/nats-io/nats-server/v2/server        26.359s
```

cc: @jnmoyne

### CODE REVIEW

**Comment by mprimi** [comment]
Failing tests are unrelated to my PR. Tried to re-trigger, and a different test failed.

File: server/jetstream_benchmark_interest_stream_limit_test.go [hunk 1]
```diff
@@ -0,0 +1,310 @@
+// Copyright 2022 The NATS Authors
```
Comment (derekcollison) [inline server/jetstream_benchmark_interest_stream_limit_test.go:1]: Wrong year.

File: server/jetstream_benchmark_interest_stream_limit_test.go [hunk 2]
```diff
@@ -0,0 +1,310 @@
...
+	"fmt"
+	"math/rand"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/nats-io/nats.go"
+)
+
+func BenchmarkJetStreamInterestStreamWithLimit(b *testing.B) {
```
Comment (derekcollison) [inline server/jetstream_benchmark_interest_stream_limit_test.go:29]: Should we move these over to test directory, starting to get crowded in this directory with very long named tests etc.

Comment (derekcollison) [reply server/jetstream_benchmark_interest_stream_limit_test.go:29]: Do we have a sense of the private server functions they need access to?

Comment (mprimi) [reply server/jetstream_benchmark_interest_stream_limit_test.go:29]: Sorry that was ambiguous. No private server symbols are used.
Only private test helpers such as `RunBasicJetStreamServer` or `createJetStreamClusterExplicit`.

I'll try to move these benchmarks, maybe it's easier than i expect.

**Comment by mprimi** [comment]
@derekcollison I have a patch that merges `jetstream_chaos_*_test.go` and `and jetstream_benchmark_*_test.go` into two files (down from 8). 

Mind if i send that separately once this is merged (I don't want the 3 to be squashed, hard to cherry-pick!)

**Comment by derekcollison** [comment]
That sounds great.

**Comment by mprimi** [comment]
Any feedback on this benchmark? If not, can you approve/merge?

**Comment by derekcollison** [comment]
I thought we were doing another that moved things around and this would be closed.

**Review by derekcollison — APPROVED** [review]
LGTM

**Comment by mprimi** [comment]
Test error is unrelated to my changes.
Can you merge when you have a second (no squashing if possible, for cherry-pick's sake)

**Pull request merged**